sparse_ho/algo/forward.py

```diff
@@ -127,9 +127,14 @@ def get_beta_jac_iterdiff(
             if verbose:
                 print("relative decrease = ", (pobj[-2] - pobj[-1]) / pobj0)
 
-        if use_stop_crit:
-            if (i > 1) and (pobj[-2] - pobj[-1] <= np.abs(pobj0 * tol)):
-                break
+        if use_stop_crit and i % Tgap == 0 and i > 0:
```

Comment:
is the use_stop_crit parameter really useful, compared to passing tol = 0 when you want to indeed run to max_iter?

Reply:
Now that there is the gap this should be equivalent  indeed, I will remove it in another pr